### ISSUE TITLE
cert-manager RBAC not updated on `eksctl upgrade`

### ISSUE THREAD

**Issue by smarsh-tim** [aws/eks-anywhere#1572]
**What happened**:
My EKS Anywhere cluster was running on an older version of Kubernetes, which was bundled with this version of cert-manager:
```
public.ecr.aws/eks-anywhere/jetstack/cert-manager-controller:v1.1.0-eks-a-1
```

I then upgraded my EKS Anywhere cluster to 1.21.5 using `eksctl upgrade`, which bumped me to:
```
public.ecr.aws/eks-anywhere/jetstack/cert-manager-controller:v1.5.3-eks-a-6
```

However, none of the RBAC changes were added in. This newer version of cert-manager has quite a few changes in RBAC, so I started seeing unexpected behavior in certificate rotations - and errors in my cert-manager logging.

**What you expected to happen**:
I would expect when `eksctl upgrade` updates the cert-manager version in the deployment, that it also updates the RBAC rules.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- EKS Anywhere Release: 
```sh
% eksctl anywhere version
v0.7.0
```
- EKS Distro Release: `1.21.5`

**Comment by ptrivedi** [comment]
@smarsh-tim in an effort to reproduce the issue, I tried this upgrade from 1.20 to 1.21 and ensured that the manifest that goes with the newer version of the cert-manager and has the rbac changes gets applied correctly to the cluster during upgrade. Do you mind sharing what logs and errors you saw? Also, I am assuming you used the `eksctl anywhere upgrade cluster -f <cluster-config>.yaml` command to perform this upgrade?

**Comment by smarsh-tim** [comment]
I don't have the logs anymore, since I short-term fixed it already. I implemented a work-around by grabbing all the RBAC settings that get installed on a fresh EKSA cluster, and I added those manifests to my GitOps repository.

In my case, I did have already a hard-coded manifest for the cert-manager deployment - as I needed to mount some certificate authority certs to that Deployment. But I didn't have hard-coded any RBAC values, so I was surprised that during the upgrade it didn't update RBAC.  I wonder if my hard-coded Deployment somehow borked the RBAC changes getting applied.

When I completed the upgrade, there were no errors and it claimed to have upgraded cert-manager. But looking at the RBAC rules they were missing some core pieces - like leases permissions, used to establish leader position for the cert-manager-controller.

Even on an upgrade from 1.21 -> 1.21, but just a newer eksRelease version (e.g. 8 instead of 7), it didn't attempt to reconcile the cert-manager settings either.

Is it possible to have any `eksctl upgrade` command reconcile / re-apply the expected manifests? Or, instead of having them one-time applied via `eksctl upgrade`, that those manifests be exposed and added to my GitOps repository so that they're consistently reconciled along with all other manifests?
